src/compositemark/boxplot.ts

```diff
@@ -42,78 +45,60 @@ export const VL_ONLY_BOXPLOT_CONFIG_PROPERTY_INDEX: {
   const {size: size, ...nonPositionEncodingWithoutSize} = nonPositionEncoding;
   const {color: _color, ...nonPositionEncodingWithoutColorSize} = nonPositionEncodingWithoutSize;
   const midTickAndBarSizeChannelDef = size ? {size: size} : {size: {value: config.box.size}};
 
-  let discreteAxisFieldDef;
-  let continuousAxisChannelDef: PositionFieldDef<Field>;
-  let discreteAxis;
-  let continuousAxis;
+  let kIQRScalar: number = undefined;
+  if (isBoxPlotDef(mark)) {
+    if (mark.extent) {
+      if(isNumber(mark.extent)) {
+        kIQRScalar = mark.extent;
+      }
+    }
+  }
+  const isMinMax = kIQRScalar === undefined;
 
-  if (isFieldDef(encoding.x) && isFieldDef(encoding.y)) {
-    // 2D
-
-    const orient: Orient = box2DOrient(spec);
-    const params = box2DParams(spec, orient);
-    discreteAxisFieldDef = params.discreteAxisFieldDef;
-    continuousAxisChannelDef = params.continuousAxisChannelDef;
-    discreteAxis = params.discreteAxis;
-    continuousAxis = params.continuousAxis;
-
-  } else if (isFieldDef(encoding.x) && isContinuous(encoding.x) && encoding.y === undefined) {
-    // 1D horizontal
-    continuousAxis = 'x';
-    continuousAxisChannelDef = encoding.x;
-  } else if (encoding.x === undefined && isFieldDef(encoding.y) && isContinuous(encoding.y)) {
-    // 1D vertical
-    continuousAxis = 'y';
-    continuousAxisChannelDef = encoding.y;
-  } else {
-    throw new Error('Need a continuous axis for 1D boxplots');
+  if (encoding.x === undefined && encoding.y === undefined) {
+    throw new Error('Need at least one axis');
   }
 
+  const orient: Orient = boxOrient(spec);
+  const {discreteAxisChannelDef, continuousAxisChannelDef, discreteAxis, continuousAxis, is1D} = boxParams(spec, orient);
+
   if (continuousAxisChannelDef.aggregate !== undefined && continuousAxisChannelDef.aggregate !== BOXPLOT) {
     throw new Error(`Continuous axis should not have customized aggregation function ${continuousAxisChannelDef.aggregate}`);
   }
 
-  const baseContinuousFieldDef = {
-      field: continuousAxisChannelDef.field,
-      type: continuousAxisChannelDef.type
-  };
-
-  const minFieldDef = {
-    aggregate: 'min',
-    ...baseContinuousFieldDef
-  };
-  const minWithAxisFieldDef = {
-    axis: continuousAxisChannelDef.axis,
-    ...minFieldDef
-  };
-  const q1FieldDef = {
-    aggregate: 'q1',
-    ...baseContinuousFieldDef
-  };
-  const medianFieldDef = {
-    aggregate: 'median',
-    ...baseContinuousFieldDef
-  };
-  const q3FieldDef = {
-    aggregate: 'q3',
-    ...baseContinuousFieldDef
-  };
-  const maxFieldDef = {
-    aggregate: 'max',
-    ...baseContinuousFieldDef
-  };
-
-  const discreteAxisEncodingMixin = discreteAxisFieldDef !== undefined ? {[discreteAxis]: discreteAxisFieldDef} : {};
+  const transform = boxTransform(encoding, discreteAxisChannelDef, continuousAxisChannelDef, kIQRScalar, is1D);
+  const {x: _xTemp, y: _yTemp, ...nonPositionEncoding2} = encoding;
```

Comment:
I don't think I understand this but this sounds super complicated and I don't like the fact that you claim that you have a "hack" here. 

1) why `_xTemp` -- is it temporary in any way -- it also does not seem used.  

2) What is `nonPositionEncoding2`?  Can you name it in a more descriptive way?  This will definitely confuse people.